src/hugchat/prompt.py

```diff
@@ -0,0 +1,674 @@
+# Pre-defined Prompts for the hugchat command line interface.
+
+def get_roles():
+    """Return a list containing just the names of all defined roles."""
+    return [role['role'] for role in PROMPTS.values()]
+
+def handle_prompt():
+    # Print a menu allowing users to select a role
+    print("\nAvailable Roles:\n")
+    for index, role in enumerate(get_roles(), start=1):
+        print(f"[{index}.{role}]", end=", ")
+
+    # Request the user's selection
+    choice = None
+    while not isinstance(choice, int) or choice < 1 or choice > len(PROMPTS):
+        try:
+            choice = int(input("\nPlease make a valid selection: ")) - 1
+        except ValueError:
+            pass
+
+    # Obtain the selected role and associated details
+    chosen_role = get_roles()[choice]
+    prompt = next((p for p in PROMPTS.values() if p['role'].lower() == chosen_role.lower()), None)
+
+    if prompt:
+        role = prompt['role']
+        description = prompt['description']
+        return role, description
+
+    raise Exception(f"No such role found ({chosen_role})")
+
+# Prompt list (in dict)
+PROMPTS ={
```

Comment:
It also looks like a few of these prompts are from https://github.com/f/awesome-chatgpt-prompts. It is creative commons, but it would likely be nice to reference them.

Reply:
> It also looks like a few of these prompts are from https://github.com/f/awesome-chatgpt-prompts. It is creative commons, so it would be good to reference them.

i was'nt sure whom to reference  the repo or the indivisuals , cz each prompt has one